examples/docker/example-package.json

```diff
@@ -27,9 +30,8 @@
     "@theia/preferences": "1.43.1",
     "@theia/process": "1.43.1",
     "@theia/terminal": "1.43.1",
-    "@theia/vsx-registry": "1.43.1",
     "@theia/workspace": "1.43.1",
-    "theia-traceviewer": "next"
+    "theia-traceviewer": "0.2.0-next.20231206220005.ff0943f.0"
```

Comment:
Have you considered using `"next"` as version? This way it would not be necessary to update that `package.json` as often. If there's even an issue with the latest `next`, one could fix it with a PR, merge it and after it's published rebuild the Docker image.

Reply:
Or leaving it as "next" I guess

Reply:
I see a related  mention in the description: 

> the docker image generated is stable and reproducible, not based on "next"

Is that referring to the `theia-traceviewer` extension or using the specific version of the base Docker image? Stability has been an issue?

Reply:
I'll try to answer all 3 comments at once, please ping if it is not clear.

Regarding the `"next"` questions:

The goal of this "example/docker" is to provide an example of how a user could create an image of the theia-trace-extension. I think that if the user plans to build and use such image in a stable and controlled product, the user should be able to have a proper way to control which version of the theia-trace-extension he/she is using. Using `"next"` would mean that every time the user builds a image, a new theia-trace-extension version could be included. Hence it would be non reproducible. Note also that:
- with the proposed change, it is not needed to update this docker at all, apart if we want to keep it up to date with the latest and greatest
- the project is not pushing images anywhere

Regarding the "stable and reproducible" topic (which I partly touched above):

Maybe here I used some terms that could be misunderstood. But "reproducible" is connected to the concept above (i.e. not using next) so that a developer could use git history to go back to a dockerfile that generated a specific image. The "stable" term is related to the image build process. Before this commit I often got issues with "vscode-ripgrep" while building the image, due to API rate limit exceed. After this commit, I have not seen this issue so often any longer, and I suspect it is due to the removal of the vsx-registry (which is however not needed for this example)

Reply:
@frallax thank you for the response. I guess we have he a trade-off between maintenance effort and stability of included building-blocks. I'm ok with what you propose. Note that no matter what, having a `npm` app, we will never have a 100% certainty the same version of every dependency will be used, from one day to the next. This is doubly true in this case where we do not preserve the `yarn.lock` when building the image. 

> Before this commit I often got issues with "vscode-ripgrep" while building the image, due to API rate limit exceed

I can confirm that `@vscode/ripgrep` is no longer pulled, now that `@theia/vsx-registry` is removed. The dependency was indirect, through the Theia plugin system. So long as the resulting Docker appliance does not require running vscode extensions, we're fine.